### PR TITLE
allow flag to disable matching variants

### PR DESCRIPTION
**Pull request by schleichardt**
@lauraluiz please review, but it is not urgent

### CODE REVIEW

File: commercetools-models/src/main/java/io/sphere/sdk/producttypes/ProductTypeDraft.java [hunk 1]
```diff
@@ -14,8 +14,7 @@
  */
 @JsonDeserialize(as = ProductTypeDraftDsl.class)
 @ResourceDraftValue(factoryMethods = {
-        @FactoryMethod(parameterNames = {"key", "name", "description", "attributes"}),
-        @FactoryMethod(methodName = "ofKey", parameterNames = {"key"})
```
Comment (lauraluiz) [inline commercetools-models/src/main/java/io/sphere/sdk/producttypes/ProductTypeDraft.java]: I guess is not important for this PR, but why did you remove the ofKey?

Comment (schleichardt) [inline commercetools-models/src/main/java/io/sphere/sdk/producttypes/ProductTypeDraft.java]: At the creation of this generated factory method I thought `key` is the only mandatory parameter. Then I used it and got exceptions due to missing data for product types. I allowed myself to remove this method again since it has not been released yet.

**Comment by schleichardt** [comment]
@lauraluiz Can you check also the last commit?

**Comment by lauraluiz** [comment]
Looks good! But difficult to tell if you missed any :D
